overlord/snapstate/snapmgr.go

```diff
@@ -107,6 +107,11 @@ type SnapSetup struct {
 	// ~/.snap/data in the current change. So a 'false' value doesn't mean the
 	// dir isn't hidden. This prevents us from always having to set it.
 	MigratedHidden bool `json:"migrated-hidden,omitempty"`
+
+	// MigratedExposed is set if the migration to a hidden snap dir was undone in
+	// the current change. A 'false' value doesn't mean the dir is hidden, just
+	// that it wasn't exposed in this change.
+	MigratedExposed bool `json:"migrated-exposed,omitempty"`
```

Comment:
the code in https://github.com/snapcore/snapd/blob/ba97d60445d52dd454cd39bce9450107de82411a/overlord/snapstate/handlers.go#L3635

probably needs a remark why this other flag is not something to consider there

Reply:
Thank you for pointing this out, I think it makes sense to consider it actually. It won't affect the rest of the change, since there are no tasks afterwards which it could affect, but I think it makes the code more consistent and robust.
Similarly, I also reset the SnapSetup flags after undoing. It won't change execution since we don't use them after an undo but it makes everything consistent.